catch exception ReadTimeoutError for Acoustic send

## catch exception `ReadTimeoutError` for Acoustic send
During the Aug 4, 2022 Acoustic incident, the Papertrail logs (below) were captured off and a new uncaught exception was seen. 

> Aug 04 17:54:20 2022-08-04 17:54:20,028 [ERROR] requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='transact-campaign-us-6.goacoustic.com', port=443): Read timed out. (read timeout=10)
> Aug 04 17:54:20 urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='transact-campaign-us-6.goacoustic.com', port=443): Read timed out. (read timeout=10)
> Aug 04 17:54:20 acoustic_tx.send_mail(
> Aug 04 17:54:20 File "/app/donate/utility/acoustic/acoustic.py", line 138, in send_mail
> Aug 04 17:54:20 File "/app/donate/utility/acoustic/acoustic.py", line 115, in attempt_send_mail


## Expected behavior
Capture the exception `ReadTimeoutError` on line [140 of acoustic.py ](https://github.com/mozilla/donate-wagtail/blob/9b3d7b9b803f30d6db9352cb638083a8d82e6832/donate/utility/acoustic/acoustic.py#L140).  
For example:
`except (ConnectionError, ReadTimeoutError) as err:`

Also capture the exception message  (the `as err` above) and print it out if retries fail (in the else statement) so we continue to capture the context around the exceptions & since we will now processing more than one exception.

--- a/donate/utility/acoustic/acoustic.py
+++ b/donate/utility/acoustic/acoustic.py
@@ -144,7 +144,7 @@ class AcousticTransact(Silverpop):
                     logger.error(
                         f"Could not send email receipt. Unable to connect to Acoustic after {send_retries} retries."
                     )
-                    logger.error(err)
+                    logger.error(f"Error: {err}")
 
                 continue
 

--- a/donate/utility/acoustic/acoustic.py
+++ b/donate/utility/acoustic/acoustic.py
@@ -5,7 +5,7 @@ from django.conf import settings
 from django.utils.encoding import force_bytes
 
 from lxml import etree
-from requests import ConnectionError
+from requests import ConnectionError, ReadTimeout
 from silverpop.api import Silverpop, SilverpopResponseException
 
 
@@ -137,13 +137,14 @@ class AcousticTransact(Silverpop):
                 # explicitly block, just in case the implicit behaviour changes in the future
                 schedule.run(blocking=True)
                 break
-            except ConnectionError:
+            except (ConnectionError, ReadTimeout) as err:
                 if attempt != 2:
-                    logger.error("Error connecting to Acoustic to send email receipt. Trying again.")
+                    logger.error("Error connecting to Acoustic while sending email receipt. Trying again.")
                 else:
                     logger.error(
-                        f"Could not send email receipt. Unable to connect to Acoustic {send_retries} retries."
+                        f"Could not send email receipt. Unable to connect to Acoustic after {send_retries} retries."
                     )
+                    logger.error(err)
 
                 continue
 

